core/drivers/atmel_trng.c

```diff
@@ -51,13 +45,12 @@ static uint32_t atmel_trng_read32(void)
 	return value;
 }
 
-TEE_Result crypto_rng_read(void *buf, size_t len)
+TEE_Result hw_get_random_bytes(void *buf, size_t len)
 {
 	uint8_t *rngbuf = buf;
 	uint32_t val = 0;
 	size_t len_to_copy = 0;
 
-	assert(buf);
```

Comment:
why removing this assertion?

Reply:
We already check the buffer before calling `hw_get_random_bytes()`, it is guaranteed to be valid by the caller at this point.